plugins/testPlugin/testPlugin.cpp

```diff
@@ -0,0 +1,475 @@
+mumble_connection_t activeConnection;
+mumble_plugin_id_t ownID;
+
+//////////////////////////////////////////////////////////////
+//////////////////// OBLIGATORY FUNCTIONS ////////////////////
+//////////////////////////////////////////////////////////////
+// All of the following function must be implemented in order for Mumble to load the plugin
+
+mumble_error_t mumble_init(uint32_t id) {
+	pluginLog("Initialized plugin");
+
+	ownID = id;
+
+	// Print the connection ID at initialization. If not connected to a server it should be -1.
+	pLog() << "Plugin ID is " << id << std::endl;
+
+	mumAPI.log(ownID, "Intitialized");
+
+	// Little showcase for how to retrieve a setting from Mumble
+	int voiceHold;
+	mumble_error_t error = mumAPI.getMumbleSetting_int(ownID, MSK_AUDIO_INPUT_VOICE_HOLD, &voiceHold);
+	if (error == STATUS_OK) {
+		pLog() << "The voice hold is set to " << voiceHold << std::endl;
+	} else {
+		pluginLog("Failed to retrieve voice hold");
+		pLog() << errorMessage(error) << std::endl;
+	}
+
+	// STATUS_OK is a macro set to the appropriate status flag (ErrorCode)
+	// If you need to return any other status have a look at the ErrorCode enum
+	// inside PluginComponents.h and use one of its values
+	return STATUS_OK;
+}
+
+void mumble_shutdown() {
+	pluginLog("Shutdown plugin");
+
+	mumAPI.log(ownID, "Shutdown");
+}
+
+MumbleStringWrapper mumble_getName() {
+	static const char *name = "TestPlugin";
+
+	MumbleStringWrapper wrapper;
+	wrapper.data = name;
+	wrapper.size = strlen(name);
+	// It's a static String and therefore doesn't need releasing
+	wrapper.needsReleasing = false;
+
+	return wrapper;
+}
+
+mumble_version_t mumble_getAPIVersion() {
+	// MUMBLE_PLUGIN_API_VERSION will always contain the API version of the used header file (the one used to build
+	// this plugin against). Thus you should always return that here in order to no have to worry about it.
+	return MUMBLE_PLUGIN_API_VERSION;
+}
+
+void mumble_registerAPIFunctions(void *api) {
+	// In this function the plugin is presented with a struct of function pointers that can be used
+	// to interact with Mumble. Thus you should store it somewhere safe for later usage.
+
+	// The pointer has to be cast to the respective API struct. You always have to cast to the same API version
+	// as this plugin itself is using. Thus if this plugin is compiled using the API version 1.0.x (where x is an arbitrary version)
+	// the pointer has to be cast to MumbleAPI_v_1_0_x.
+	// Furthermore the struct HAS TO BE COPIED!!! Storing the pointer is not an option as it will become invalid quickly!
+
+	// **If** you are using the same API version that is specified in the included header file (as you should), you
+	// can simply use the MUMBLE_API_CAST to cast the pointer to the correct type and automatically dereferencing it.
+	mumAPI = MUMBLE_API_CAST(api);
+
+	pluginLog("Registered Mumble's API functions");
+}
+
+void mumble_releaseResource(const void *pointer) {
+	throw "";
```

Comment:
```suggestion
	std::cerr << "[ERROR]: Unexpected call to mumble_releaseResources" << std::endl;
	std::terminate();
```
Don't throw from a C function!